lib/classes/Swift/Mime/IdGenerator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of SwiftMailer.
+ * (c) 2004-2009 Chris Corbyn
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Egulias\EmailValidator\EmailValidator;
+
+/**
+ * Message ID generator.
+ */
+class Swift_Mime_IdGenerator implements Swift_IdGenerator
+{
+    /**
+     * @param EmailValidator $emailValidator
+     * @param string|bool    $idRight
+     */
+    public function __construct(EmailValidator $emailValidator, $idRight = false)
+    {
+        if ($idRight) {
+            $this->idRight = $idRight;
```

Comment:
In the existing code, the validation is only used to check if the contents of $_SERVER['SERVER_NAME'] contains something sane (AFAICT it always does, unless it has been filled with garbage on purpose).

The ID will be validated when the message is sent (in Swift_Mime_Headers_IdentificationHeader). It seems most in line with the behaviour elsewhere in Swift to throw errors on send, not on setting values or instantiating objects, so I think we should leave this. But I do not have strong opinions about this.